rust/rpc/src/lib.rs

```diff
@@ -222,13 +225,17 @@ impl<W: Write + Send> RpcLoop<W> {
             scope.spawn(move|| {
                 let mut stream = rf();
                 loop {
+                    let json_decode = xi_trace::trace_block("decode rpc", &["rpc", "json"]);
```

Comment:
This block will be spending most of it's time waiting on new input, we probably want to be tracing in `rpc/src/parse.rs`. Get that this PR is mostly intended as an illustration, though.

Reply:
Yeah, this CL is just an illustration not intended for merging.  Knowing how much time the process is waiting on input isn't actually a terrible idea but you're right the RPC decoding duration is better measured elsewhere.  I can close the PR to make it more clear if you feel the example is OK to find in the closed PRs.

Reply:
I don't mind keeping it open until something gets merged, up to you. :)